### PR TITLE
Audit/Tests for Go Leaks

### PR DESCRIPTION
**Pull request by mattlord**
## Description

I did a quick audit of places where we were using a confined Ticker — created as a local variable in a function — but not ensuring it gets stopped with a defer statement. This will help to prevent resource leaks. This was prompted by the fact I realized I forgot to do this in a PR yesterday: https://github.com/vitessio/vitess/pull/10720

I'm also investigating usage of this: https://github.com/uber-go/goleak
Unfortunately this does not seem useful to use in e2e tests.

## Related Issue(s)

  - Follow-up to: https://github.com/vitessio/vitess/pull/10720

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/tabletserver/tabletserver.go [hunk 1]
```diff
@@ -349,6 +349,7 @@ func (tsv *TabletServer) InitACL(tableACLConfigFile string, enforceTableACLConfi
 
 	if reloadACLConfigFileInterval != 0 {
 		ticker := time.NewTicker(reloadACLConfigFileInterval)
+		defer ticker.Stop()
```
Comment (rohit-nayak-ps) [inline go/vt/vttablet/tabletserver/tabletserver.go:352]: Don't we want this timer to keep running in the background to reload the acl config every `reloadACLConfigFileInterval` seconds?

Comment (mattlord) [reply go/vt/vttablet/tabletserver/tabletserver.go:352]: Hmm, you're right! Good catch! We have a never ending for loop below but that's in a goroutine so we don't want to stop the ticker when this outer function ends. Will change.

File: go/mysql/collations/tools/makecolldata/contractions.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func printContraction1(g *codegen.Generator, wa *weightarray, incont []uca.Contr
 			trie[r] = append(trie[r], cont)
 		}
 		if depth == len(cont.Path) {
-			leaf = &cont
+			leaf = &cont // nolint:exportloopref
```
Comment (deepthi) [inline go/mysql/collations/tools/makecolldata/contractions.go:61]: Fixed differently by @dbussink in #10765

**Review by deepthi — COMMENTED** [review]
This looks good in principle.
What did you mean by the `goleak` library not being useful in e2e tests?

**Comment by mattlord** [comment]
> This looks good in principle.
> What did you mean by the `goleak` library not being useful in e2e tests?

Seems that it only tests the main command go test executes whereas the e2e tests start up a bunch of processes.

**Pull request closed without merging**